### PR TITLE
Assemble savings charts

### PR DESCRIPTION
**Pull request by EmilMalanczak**

### CODE REVIEW

File: packages/app/src/features/savings/components/savings-charts/UsdsSavingsCharts.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  savingsRateInfo: UseSavingsRateInfoResult
+}
+
+export function UsdsSavingsCharts({
+  selectedTimeframe,
+  setSelectedTimeframe,
+  myEarningsInfo,
+  savingsRateInfo,
+}: UsdsSavingsChartsProps) {
+  const shouldDisplayMyEarnings =
```
Comment (yivlad) [inline packages/app/src/features/savings/components/savings-charts/UsdsSavingsCharts.tsx:21]: Can we extract his to the top-level hook? Also I think we should just that there is at least one non-zero element

**Review by yivlad — APPROVED** [review]

File: packages/app/src/domain/savings-charts/useMyEarningsInfo/calculatePredictions.ts [hunk 2]
```diff
@@ -9,44 +9,44 @@ const SECONDS_PER_DAY = 24 * 60 * 60
...
         timestamp,
         savingsInfo,
       })
 
     case '1Y':
     case 'All':
       return calculatePredictionsIncomeByDays({
-        days: Math.min(optimalPredictionsLength, 60),
-        balance,
+        // setting upper bounds due to visible performance problems
```
Comment (krzkaczor) [inline packages/app/src/domain/savings-charts/useMyEarningsInfo/calculatePredictions.ts:47]: are we recalculating thousands of points multiple times per second? 😬  we really need to prioritze fixing this ASAP.

File: packages/app/src/domain/savings-charts/useSavingsRateInfo/query.ts [hunk 3]
```diff
@@ -37,12 +37,11 @@ const savingsRateDataResponseSchema = z
...
     z.object({
       date: dateSchema,
-      rate: normalizedUnitNumberSchema,
-      dsr_rate: normalizedUnitNumberSchema,
+      dsr_rate: percentageSchema,
       ssr_rate: z
         .string()
         .nullable()
-        .transform((value) => (value ? NormalizedUnitNumber(value) : null)),
+        .transform((value) => (value ? Percentage(value, true) : null)),
```
Comment (krzkaczor) [inline packages/app/src/domain/savings-charts/useSavingsRateInfo/query.ts:44]: According to used validators `ssr_rate` can be above 1 but `dsr_rate` no. Why?

File: packages/app/src/domain/savings-charts/useMyEarningsInfo/calculatePredictions.ts (hunk 2 above)
Comment (EmilMalanczak) [reply packages/app/src/domain/savings-charts/useMyEarningsInfo/calculatePredictions.ts:47]: its memoized - performance meant in the comment is related to overhead while calculating large amount of predictions at once

**Comment by EmilMalanczak** [comment]
All graph stories got affected by non-type safety on stories. 
I think we should update the guideline, current guide preserves only type safety for the values but it doesn't ensure that we pass all required props 

image

image

**Review by krzkaczor — APPROVED** [review]

**Pull request merged**